dbms/src/Interpreters/tests/expression_analyzer.cpp

```diff
@@ -35,11 +35,9 @@ struct TestEntry
     {
         ASTPtr ast = parse(query);
 
-        ExpressionAnalyzer analyzer(ast, context, {}, source_columns, required_result_columns);
+        auto res =  SyntaxAnalyzer().analyze(ast, context, {}, source_columns, required_result_columns);
```

Comment:
It would be more self-documented to pass more args in SyntaxAnalyzer's ctor and call analyze with AST only:
SyntaxAnalyzer(context, {}, source_columns, required_result_columns).analyze(ast);
or
SyntaxAnalyzer(context, {}).analyze(ast, source_columns, required_result_columns);

If we don't expect changes of context and storage, we'd better set them in ctor to explain it.